x/merkledb/db.go

```diff
@@ -83,7 +83,7 @@ type ChangeProofer interface {
 		ctx context.Context,
 		proof *ChangeProof,
 		start []byte,
-		end []byte,
+		end Maybe[[]byte],
```

Comment:
Should we also use `end Maybe[[]byte]` in `GetChangeProof` above and `GetRangeProofAtRoot` below for consistency?

Reply:
Part 2